runtime/parachains/src/inclusion_inherent.rs

```diff
@@ -127,6 +136,43 @@ decl_module! {
 	}
 }
 
+/// Limit the number of backed candidates processed in order to stay within block weight limits.
+///
+/// Use a configured assumption about the weight required to process a backed candidate and the
+/// current block weight as of the execution of this function to ensure that we don't overload
+/// the block with candidate processing.
+///
+/// Retains the first N candidates which fit, with no attempt at sorting by priority.
+fn limit_backed_candidates<T: Config>(backed_candidates: &mut Vec<BackedCandidate<T::Hash>>) {
```

Comment:
Why don't we drop all or nothing?

If the backed candidates are sorted, this could lead to unfair situations.

Reply:
I assumed it was desirable to process some backed candidates, not all or nothing. Agree that if we don't trust the provisioner, then all or nothing is more fair. Can we assume we trust the provisioner?

Reply:
I would say no.

Reply:
63b4869a